tools/global_bss_check.py

```diff
@@ -0,0 +1,216 @@
+
+                # Find file
+                if entryMatch is not None:
+                    name = "/".join(entryMatch["name"].split("/")[1:])
+
+                    # mapfile only contains .o files, so just strip the last character to replace it
+                    # we assume all the .c files are in the src folder, and all others are .s (true for OoT/MM)
+                    if name.split("/")[0] == "src":
+                        name = name[:-1] + "c"
+                    else:
+                        name = name[:-1] + "s"
+
+                    size = int(entryMatch["size"], 16)
+                    vram = int(entryMatch["vram"], 16)
+
+                    if size > 0:
+                        inFile = True
+                        currentFile = name
+
+    return symbolsDict
+
+
+Compared = collections.namedtuple("Compared", [ "buildAddress", "buildFile", "expectedAddress", "expectedFile", "diff"])
+
+def compareMapFiles(mapFileBuild: str, mapFileExpected: str):
+    badFiles = set()
+    missingFiles = set()
+
+    print("Build mapfile:    " + mapFileBuild, file=os.sys.stderr)
+    print("Expected mapfile: " + mapFileExpected, file=os.sys.stderr)
+    print("", file=os.sys.stderr)
+
+    if not os.path.exists(mapFileBuild):
+        print(f"{colorama.Fore.LIGHTRED_EX}error{colorama.Fore.RESET}: mapfile not found at {mapFileBuild}. Did you enter the correct path?", file=os.sys.stderr)
+        exit(1)
+
+    if not os.path.exists(mapFileExpected):
+        print(f"{colorama.Fore.LIGHTRED_EX}error{colorama.Fore.RESET}: expected mapfile not found at {mapFileExpected}. Is 'expected' missing or in a different folder?", file=os.sys.stderr)
+        exit(1)
+
+    buildMap = parseMapFile(mapFileBuild)
+    expectedMap = parseMapFile(mapFileExpected)
+
+    comparedDict = collections.OrderedDict()
+
+    for symbol in buildMap:
+        if symbol in expectedMap:
+            comparedDict[symbol] = Compared( buildMap[symbol].vram, buildMap[symbol].file, expectedMap[symbol].vram, expectedMap[symbol].file, buildMap[symbol].vram - expectedMap[symbol].vram )
+            if comparedDict[symbol].diff != 0:
+                badFiles.add(buildMap[symbol].file)
+                
+        else:
+            missingFiles.add(buildMap[symbol].file)
+            comparedDict[symbol] = Compared( buildMap[symbol].vram, buildMap[symbol].file, -1, "", "Unknown" )
+
+    for symbol in expectedMap:
+        if not symbol in buildMap:
+            missingFiles.add(expectedMap[symbol].file)
+            comparedDict[symbol] = Compared( -1, "", expectedMap[symbol].vram, expectedMap[symbol].file, "Unknown" )
+
+    return badFiles, missingFiles, comparedDict
+
+
+def printCsv(badFiles, missingFiles, comparedDict, printAll = True):
+    print("Symbol Name,Build Address,Build File,Expected Address,Expected File,Difference,GOOD/BAD/MISSING")
+
+    allGood = True
+    allFound = True
+    
+    problemFiles = set.union(badFiles, missingFiles)
```

Comment:
```suggestion
    problemFiles = badFiles | missingFiles
```
Or  this:
`problemFiles = badFiles.union(missingFiles)`
whichever you prefer

Reply:
Oops, looks like this isn't used anyway any more.